system/loggerd/tests/test_deleter.py

```diff
@@ -73,6 +80,39 @@ def test_delete_order(self):
       self.make_file_with_data(self.seg_format2.format(0), self.f_type),
     ])
 
+  def test_delete_both_files_and_dirs(self):
```

Comment:
let's also create lots of random nested dirs and files to really cover everything

Reply:
now generating random nested dirs and files